diff --git a/dist/index.js b/dist/index.js
index 55f84c2ba25c79d84ae30730af7d54d799daa502..938c3559c59cdd01d0f1e689c07f85629247e5f4 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -18,7 +18,8 @@ function src_default({ include, exclude, compat, devtools } = {}) {
           include,
           exclude,
           babel: {
-            cwd: fileURLToPath(babelCwd)
+            cwd: fileURLToPath(babelCwd),
+            plugins: [[ "@babel/plugin-proposal-decorators", { "version": "2023-11" } ]]
           }
         });
         const viteConfig = {
