diff --git a/dist/emotion-cache.cjs.js b/dist/emotion-cache.cjs.js
index 7d0bec242f31aed585ca52217f82225a5c44a308..31c619c1120e8a310e830465787e0e48d7d41c3e 100644
--- a/dist/emotion-cache.cjs.js
+++ b/dist/emotion-cache.cjs.js
@@ -408,7 +408,7 @@ var createCache = function createCache(options) {
 
   var _insert;
 
-  var omnipresentPlugins = [compat, removeLabel];
+  var omnipresentPlugins = [removeLabel];
 
   if (!getServerStylisCache) {
     var currentSheet;
