diff --git a/dist/core/build/plugins/plugin-css.js b/dist/core/build/plugins/plugin-css.js
index 839ab99dd2b8c743bd7f635216a4ed2b692ba34c..82002de26bdc4289db671ee7760527a401ea8a65 100644
--- a/dist/core/build/plugins/plugin-css.js
+++ b/dist/core/build/plugins/plugin-css.js
@@ -10,7 +10,7 @@ import { getPageDataByViteID, getPageDatasByClientOnlyID } from "../internal.js"
 import { extendManualChunks, shouldInlineAsset } from "./util.js";
 function pluginCSS(options, internals) {
   return {
-    targets: ["client", "server"],
+    targets: ["server"],
     hooks: {
       "build:before": ({ target }) => {
         let plugins = rollupPluginAstroBuildCSS({
