diff --git a/dist/core/build/plugins/plugin-css.js b/dist/core/build/plugins/plugin-css.js
index 43088c2470ced954dbd6c1ed3e603d649b3d56e1..46fd543e971e12fd1591375588ed1760c67dfa44 100644
--- a/dist/core/build/plugins/plugin-css.js
+++ b/dist/core/build/plugins/plugin-css.js
@@ -14,7 +14,7 @@ import {
 import { extendManualChunks, shouldInlineAsset } from "./util.js";
 function pluginCSS(options, internals) {
   return {
-    targets: ["client", "server"],
+    targets: ["server"],
     hooks: {
       "build:before": ({ target }) => {
         let plugins = rollupPluginAstroBuildCSS({
