diff --git a/package.json b/package.json
index 3fa17f13652e65ab8de2fba87aa9c469c2eed372..c8537ceb9b635e654b2cb146bf8e175a3910962f 100644
--- a/package.json
+++ b/package.json
@@ -12,56 +12,56 @@
   "exports": {
     ".": {
       "types": "./src/index.d.ts",
-      "browser": "./dist/preact.module.js",
+      "browser": "./src/index.js",
       "umd": "./dist/preact.umd.js",
       "import": "./dist/preact.mjs",
       "require": "./dist/preact.js"
     },
     "./compat": {
       "types": "./compat/src/index.d.ts",
-      "browser": "./compat/dist/compat.module.js",
+      "browser": "./compat/src/index.js",
       "umd": "./compat/dist/compat.umd.js",
       "import": "./compat/dist/compat.mjs",
       "require": "./compat/dist/compat.js"
     },
     "./debug": {
       "types": "./debug/src/index.d.ts",
-      "browser": "./debug/dist/debug.module.js",
+      "browser": "./debug/src/index.js",
       "umd": "./debug/dist/debug.umd.js",
       "import": "./debug/dist/debug.mjs",
       "require": "./debug/dist/debug.js"
     },
     "./devtools": {
       "types": "./devtools/src/index.d.ts",
-      "browser": "./devtools/dist/devtools.module.js",
+      "browser": "./devtools/src/index.js",
       "umd": "./devtools/dist/devtools.umd.js",
       "import": "./devtools/dist/devtools.mjs",
       "require": "./devtools/dist/devtools.js"
     },
     "./hooks": {
       "types": "./hooks/src/index.d.ts",
-      "browser": "./hooks/dist/hooks.module.js",
+      "browser": "./hooks/src/index.js",
       "umd": "./hooks/dist/hooks.umd.js",
       "import": "./hooks/dist/hooks.mjs",
       "require": "./hooks/dist/hooks.js"
     },
     "./test-utils": {
       "types": "./test-utils/src/index.d.ts",
-      "browser": "./test-utils/dist/testUtils.module.js",
+      "browser": "./test-utils/src/index.js",
       "umd": "./test-utils/dist/testUtils.umd.js",
       "import": "./test-utils/dist/testUtils.mjs",
       "require": "./test-utils/dist/testUtils.js"
     },
     "./jsx-runtime": {
       "types": "./jsx-runtime/src/index.d.ts",
-      "browser": "./jsx-runtime/dist/jsxRuntime.module.js",
+      "browser": "./jsx-runtime/src/index.js",
       "umd": "./jsx-runtime/dist/jsxRuntime.umd.js",
       "import": "./jsx-runtime/dist/jsxRuntime.mjs",
       "require": "./jsx-runtime/dist/jsxRuntime.js"
     },
     "./jsx-dev-runtime": {
       "types": "./jsx-runtime/src/index.d.ts",
-      "browser": "./jsx-runtime/dist/jsxRuntime.module.js",
+      "browser": "./jsx-runtime/src/index.js",
       "umd": "./jsx-runtime/dist/jsxRuntime.umd.js",
       "import": "./jsx-runtime/dist/jsxRuntime.mjs",
       "require": "./jsx-runtime/dist/jsxRuntime.js"
diff --git a/src/diff/index.js b/src/diff/index.js
index ae2d77ffc2351719d4390ceec43b3edbfdaf0bb7..38b0c203b8adca160fecdc0ca51db423485a0e16 100644
--- a/src/diff/index.js
+++ b/src/diff/index.js
@@ -227,6 +227,7 @@ export function diff(
 					c.state = c._nextState;
 				} while (c._dirty && ++count < 25);
 			}
+			options._rendered?.(newVNode);
 
 			// Handle setState called in render, see #2553
 			c.state = c._nextState;
@@ -462,7 +463,7 @@ function diffElementNodes(
 				) {
 					continue;
 				}
-				setProperty(dom, i, null, value, namespace);
+				setProperty(dom, newVNode, i, null, value, namespace);
 			}
 		}
 
@@ -482,7 +483,7 @@ function diffElementNodes(
 				(!isHydrating || typeof value == 'function') &&
 				oldProps[i] !== value
 			) {
-				setProperty(dom, i, value, oldProps[i], namespace);
+				setProperty(dom, newVNode, i, value, oldProps[i], namespace);
 			}
 		}
 
@@ -546,12 +547,12 @@ function diffElementNodes(
 					// again, which triggers IE11 to re-evaluate the select value
 					(nodeType === 'option' && inputValue !== oldProps[i]))
 			) {
-				setProperty(dom, i, inputValue, oldProps[i], namespace);
+				setProperty(dom, newVNode, i, inputValue, oldProps[i], namespace);
 			}
 
 			i = 'checked';
 			if (checked !== undefined && checked !== dom[i]) {
-				setProperty(dom, i, checked, oldProps[i], namespace);
+				setProperty(dom, newVNode, i, checked, oldProps[i], namespace);
 			}
 		}
 	}
diff --git a/src/diff/props.js b/src/diff/props.js
index 316146f0fdfb91456047866931a02ba7bd8c0b51..c35387836c9e9594aebfbdc2ad73b84db57241bd 100644
--- a/src/diff/props.js
+++ b/src/diff/props.js
@@ -29,12 +29,13 @@ let eventClock = 0;
 /**
  * Set a property value on a DOM node
  * @param {PreactElement} dom The DOM node to modify
+ * @param {VNode} vnode The VNode corresponding to the DOM node
  * @param {string} name The name of the property to set
  * @param {*} value The value to set the property to
  * @param {*} oldValue The old value the property had
  * @param {string} namespace Whether or not this DOM node is an SVG node or not
  */
-export function setProperty(dom, name, value, oldValue, namespace) {
+export function setProperty(dom, vnode, name, value, oldValue, namespace) {
 	let useCapture;
 
 	o: if (name === 'style') {
@@ -77,7 +78,17 @@ export function setProperty(dom, name, value, oldValue, namespace) {
 		else name = name.slice(2);
 
 		if (!dom._listeners) dom._listeners = {};
-		dom._listeners[name + useCapture] = value;
+		let listener, parent = vnode;
+		listener = dom._listeners[name + useCapture] ??= {};
+		listener.f = value
+		
+		while (parent) {
+			if (parent._component && value.name in parent._component && parent._component[value.name] === value) {
+				listener.c = parent._component
+				break
+			}
+			parent = parent._parent
+		}
 
 		if (value) {
 			if (!oldValue) {
@@ -166,7 +177,7 @@ function createEventProxy(useCapture) {
 			} else if (e._dispatched < eventHandler._attached) {
 				return;
 			}
-			return eventHandler(options.event ? options.event(e) : e);
+			return eventHandler.f.call(eventHandler.c, options.event ? options.event(e) : e);
 		}
 	};
 }
