diff --git a/dist/node/chunks/dep-BJP6rrE_.js b/dist/node/chunks/dep-BJP6rrE_.js
index a7b3e26f3a24f75d26a7df6231479948e761ecbf..837fb9758024f27ef7fb464eb9ce874aa9018590 100644
--- a/dist/node/chunks/dep-BJP6rrE_.js
+++ b/dist/node/chunks/dep-BJP6rrE_.js
@@ -50090,7 +50090,7 @@ function buildImportAnalysisPlugin(config) {
       if (!imports.length) {
         return null;
       }
-      const insertPreload = getInsertPreload(this.environment);
+      const insertPreload = false;
       const dynamicImports = {};
       if (insertPreload) {
         let match;
