diff --git a/dist/node/chunks/dep-yUJfKD1i.js b/dist/node/chunks/dep-yUJfKD1i.js
index 57759ca8c9eb2362b67c078ea0504aafd888b44c..a6c09afd9e40a9a86ae2f2ad65af9559b3d7838d 100644
--- a/dist/node/chunks/dep-yUJfKD1i.js
+++ b/dist/node/chunks/dep-yUJfKD1i.js
@@ -49775,7 +49775,7 @@ function buildImportAnalysisPlugin(config) {
       if (!imports.length) {
         return null;
       }
-      const insertPreload = getInsertPreload(this.environment);
+      const insertPreload = false;
       const dynamicImports = {};
       if (insertPreload) {
         let match;
